applications/feedback_sentiment_analyzer/SPECIFICATION.md

```diff
@@ -0,0 +1,383 @@
+# FSA technical specification
+
+This document contains the technical specifications for Feedback Sentiment Analyzer (FSA), a sample application that showcases AWS services and SDKs.
+
+This document explains:
+
+- Application inputs and outputs.
+- Underlying AWS components and their configurations.
+- Key cross-service integration details.
+
+See the [README.md](README.md) for an introduction to FSA.
+
+---
+### Table of Contents
+
+  * [Diagram](#diagram)
+  * [User actions](#user-actions)
+  * [Application behavior](#application-behavior)
+  * [HTTP API specification](#http-api-specification)
+  * [Step Function configuration](#step-function-configuration)
+    + [State machine Lambda functions](#state-machine-lambda-functions)
+      + [ExtractText](#extracttext)
+      + [AnalyzeSentiment](#analyzesentiment)
+      + [TranslateText](#translatetext)
+      + [SynthesizeAudio](#synthesizeaudio)
+  * [GetFeedback Lambda function](#getfeedback-lambda-function)
+  * [Processing S3 events with EventBridge](#processing-s3-events-with-eventbridge)
+  * [Managing items in DynamoDB](#managing-items-in-dynamodb)
+  * [Other FSA material](#other-fsa-material)
+
+---
+
+## Diagram
+This diagram represents the relationships between key FSA components. 
+![architecture2.png](architecture2.png)
+
+---
+
+## User actions
+
+This application receives three inputs from the frontend.
+
+- authenticate user (uses `Cognito`)
+- load page (uses `DynamoDB:GetItems`)
+- upload image to S3 (`S3:PutObject`)
+
+## Application behavior
+
+This application produces two outputs from the backend:
+
+- put new item in the database (uses `DynamoDB:PutItem`)
+- put new synthesized audio to S3 (uses `S3:PutObject`)
+
+## User flow
+The following list describes the happy path of a frontend user.
+
+1. Authenticate via Amazon Cognito hosted UI.
+2. Upload an image with text in a non-English language.
+3. Press the refresh button after a minute.
+4. See the original image alongside a English translation and a play button.
+5. Tap the play button and listen to the synthesized audio of the English translation.
+
+---
+
+## HTTP API specification
+
+All the APIs are created by the CDK script. The endpoints are common to every language variation and do not need any additional implementation.
+
+<details>
+<summary><strong>PUT /api/media/{item}</strong></summary>
+<br/>
+
+> Create or update an object in Amazon S3. Creating or updating an image file will trigger the Step Function.
```

Comment:
Are we triggering the "Step Functions workflow" or the "Step Functions step"?

Reply:
Technically we're setting a state machine as the target of an EventBridge rule. I guess workflow makes the most sense?